Integrate all roles together for full deployment

jwflory opened the pull request:
_Optionally tagging @nlMeminger, @ct-martin, and @ritjoe for review._

---

This commit brings all existing roles together and integrates them. It
makes it possible for someone to use these roles in conjunction with
each other, such as for a server configuration. It allows us to move
forward with running it on our infrastructure.

A summation of changes:

MariaDB
-------

* Copy defaults from current MariaDB configuration

php-fpm
-------

* Minor syntax cleanup

nginx
-----

* Split configurations out into three different globs
  * **restrictions.conf**: Base set of restrictions for all sites
  * **wordpress.conf**: Base set of guidelines for any WordPress site
  * **fossrit.jwf.io**: Single installed WordPress site
* Adjust various settings with PHP

Ansible
-------

* Update nginx role task to load in new global config files

Resources
---------

Resources informing this commit:

* https://codex.wordpress.org/Nginx
* https://www.nginx.com/resources/wiki/start/topics/recipes/wordpress/

ct-martin commented on roles/nginx/templates/wordpress.conf:
```diff
@@ -1,31 +1,48 @@
+            access_log      off;
+        }
 
-	client_max_body_size 64M;
```
Are you sure your want to remove the caching? I don't care personally, but just want to make sure you didn't do this by accident since you have some more caching stuff below.

jwflory replied:
@ct-martin Actually I'm not sure how this one snuck in there. I'll go ahead and remove it. Thanks for catching!

ct-martin commented:
I would also consider if you want caching since WP can be a little bandwidth heavy from experience and you need to do some things in the nginx config for that since you're not using Apache.

All said though, that can be another issue and I'm glad you found the WP Codex link useful.

jwflory commented:
@ct-martin Thanks for catching. :+1: Commit a2fa1baf8db8b7259e3477b5888a73f334ed251f addresses your feedback. I'll aim to merge this PR tonight, to give anyone else a chance to take a look.

ct-martin commented on roles/nginx/templates/wordpress.conf:
```diff
@@ -1,31 +1,48 @@
-        include                 fastcgi_params;
-    }
+        location ~* \.(js|css|png|jpg|jpeg|gif|ico)$ {
+                expires         max;
```
@jwflory there's another part here if you want it

jwflory replied:
@ct-martin Are you suggesting to remove this section? As I understand, this is instructing clients to not set an expiration in a local cache for any of these types of files, since they are usually static.

ct-martin commented:
@jwflory not sure if it's client or server cache info due to lack of
familiarity. Wasn't sure if it was related to the other header, which you
had removed

--
Christian Martin (Phone)

On Thu, Oct 4, 2018, 9:53 AM Justin W. Flory <notifications@github.com>
wrote:

> *@jwflory* commented on this pull request.
> ------------------------------
>
> In roles/nginx/templates/wordpress.conf
> <https://github.com/FOSSRIT/infrastructure/pull/15#discussion_r222677879>:
>
> >
> -    location ~ \.php$ {
> -        try_files               $uri =404;
> -        fastcgi_split_path_info ^(.+\.php)(/.+)$;
> -        fastcgi_index           index.php;
> -        fastcgi_pass            unix:/var/run/php-fpm/wordpress.sock;
> -        fastcgi_param           SCRIPT_FILENAME
> -                                $document_root$fastcgi_script_name;
> -        include                 fastcgi_params;
> -    }
> +        location ~* \.(js|css|png|jpg|jpeg|gif|ico)$ {
> +                expires         max;
>
> @ct-martin <https://github.com/ct-martin> Are you suggesting to remove
> this section? As I understand, this is instructing clients to not set an
> expiration in a local cache for any of these types of files, since they are
> usually static.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/FOSSRIT/infrastructure/pull/15#discussion_r222677879>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AYJ2H6r0kMxoCNg6CXpX1W0M4fRyxO4pks5uhhLygaJpZM4XHUlt>
> .
>

jwflory commented:
No databases are actually created yet… need to get that added to the WordPress role. I'll work on it over the long weekend. Rebased and merging now. :clapper:
